### ISSUE TITLE
[Bug]: Move user-interaction events from non-interactive elements

### ISSUE THREAD

**Issue by okdv** [okdv/colormap#31]
## Bug Report

**Describe the bug**
Not an impacting bug but an a11y one. There are non-interactive elements, like `<li>` in the legend, that have user-interaction even handlers like on:click. This works for us but would cause issues in many a11y softwares. these events should be drilled down to a child element that is interactive like `<button>` or `<a>`

**To Reproduce**
a11y warnings thrown in `npx playwright test`

**Expected behavior**
n/a

**Actual Behavior**
n/a

**Screenshots**
n/a

**Environment (please complete the following information):**
n/a

**Additional context**
n/a

**Comment by kennedfer** [comment]
I will do this, my vscode is fullly of yellow trails lol

**Comment by okdv** [comment]
@kennedfer awesome thanks! I've assigned the issue to you

**Comment by okdv** [comment]
Hi @kennedfer were you still wanting to work on this one? the code solution was good as-is, just need to take care of the git history issues. there has been an additional release since so you will want to ensure you're fetching both merge and evelop from origin and then merging that into your feature branch

**Comment by kennedfer** [comment]
I do a new PR, can you see it?
